pilot/pkg/serviceregistry/kube/conversion.go

```diff
@@ -74,8 +74,12 @@ func convertService(svc v1.Service, domainSuffix string) *model.Service {
 
 	if svc.Spec.Type == v1.ServiceTypeExternalName && svc.Spec.ExternalName != "" {
 		external = svc.Spec.ExternalName
-		resolution = model.Passthrough
-		meshExternal = true
+		if strings.HasSuffix(external, "."+domainSuffix) {
+			resolution = model.DNSLB
+		} else {
+			resolution = model.Passthrough
+			meshExternal = true
+		}
```

Comment:
So thinking more about this, we should keep the meshExternal = true.. We should not be doing the (if the externalname is an internal service...) as it breaks the semantics for an external name service. I would remove the if block and permanently mark all externalName services as model.DNSLB as they are simply cname records anyway.